src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -1138,6 +1138,8 @@ public bool TryGetNodeInfo(string name, out CustomNodeInfo info)
                 var collapsedNode = CreateCustomNodeInstance(newId, isTestMode: isTestMode);
                 collapsedNode.X = avgX;
                 collapsedNode.Y = avgY;
+                collapsedNode.Controller.SyncWithDefinitionStart += currentWorkspace.OnSyncWithDefintionStart;
+                collapsedNode.Controller.SyncWithDefinitionEnd += currentWorkspace.OnSyncWithDefinitionEnd;
```

Comment:
Every time I look at this function, I have the urge to go *modularize* it. I was looking at if the `collapsedNode` will hold `currentWorkspace` in-memory and it seems like not the case. So all is good.